enterprise/api/v1/views.py

```diff
@@ -213,6 +215,110 @@ def course_enrollments(self, request, pk):
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course. This endpoint is
+        not transactional, in that any one or more failures will not affect other successful enrollments made within
+        the same request.
+
+        Expected params:
+            - licenses_info (list of dicts): an array of dictionaries, each containing the necessary information to
+            create a licenced enrollment for a user in a specified course. Each dictionary must contain a user email, a
+            course run key, and a UUID of the license that the learner is using to enroll with.
+            Example:
+                licenses_info: [
+                    {
+                        'email': 'newuser@test.com',
+                        'course_run_key': 'course-v1:edX+DemoX+Demo_Course',
+                        'course_mode': 'verified',
+                        'license_uuid': '5b77bdbade7b4fcb838f8111b68e18ae'
+                    },
+                    ...
+                ]
+
+        Optional params:
+            - discount (int): the percent discount to be applied to all enrollments. Defaults to 100.
+
+        Expected Return Values:
+            Success cases:
+                - All users exist and are enrolled - [], 200
+                - Some or none of the users exist but are enrolled - [], 202
+
+            Failure cases:
+                - Some or all of the users can't be enrolled, no users were enrolled -
+                    {'successes': [], 'pending': [], 'failures': []}, 409
+        """
+        enterprise_customer = self.get_object()
+        serializer = serializers.EnterpriseCustomerBulkSubscriptionEnrollmentsSerializer(
+            data=request.data,
+            context={
+                'enterprise_customer': enterprise_customer,
+                'request_user': request.user,
+            }
+        )
+        try:
+            serializer.is_valid(raise_exception=True)
+        except ValidationError:
+            return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
+
+        errors = []
+        licenses_info = serializer.validated_data.get('licenses_info')
+
+        # Default subscription discount is 100%
+        discount = serializer.validated_data.get('discount', 100.00)
+
+        # Get a list of user emails from the license info.
+        emails = {license_info['email'] for license_info in licenses_info}
+        for email in emails:
+            try:
+                validate_email_to_link(email, enterprise_customer, raise_exception=False)
+            except exceptions.ValidationError as error:
+                errors.append(error)
+
+        if errors:
```

Comment:
I think we don't want this line at 281 anymore. If there are 2 bad emails in a batch of 100 it will return the 2 bad ones but won't complete the enrollments on the other 98.

Reply:
doesn't match the work I did to allow for bad enrollments later on. 👍

Reply:
How would you like these returned along with the results of `enroll_licensed_users_in_courses`?

Reply:
We're returning a list of error strings, correct? Or are we returning a list of emails? If it's the former, something simple like 'Invalid email address: {}' as the error string, appended to the existing error list should be fine.

Reply:
Currently we are returning a dictionary of success, pendings, and failures. All arrays. I can add a `invalid_email` key to that dictionary if that works for you?

Reply:
extra field added. Let me know if you're good with naming conventions 👍